visitor template

ll06printf opened the pull request:
为enum实现访问器模式的过程宏。

ll06printf committed:
初始化提交

创建 Rust 项目

ll06printf committed:
添加依赖的第三方库

- 指定 Rustc 版本（虽然cargo只能保证最低版本）
- 添加 peg 和 clap 依赖项
- 添加一个peg语法使用示例

ll06printf committed:
基本命令行参数解析

- 添加了基本的命令行参数

ll06printf committed:
解析源文件命令行参数

ll06printf committed:
项目结构构建

构建了基本的项目结构，详细参考 `README.md`

ll06printf committed:
移除测试 peg 的 example

ll06printf committed:
切换解析库为 winnow

ll06printf committed:
把 peg 添加回来

ll06printf committed:
简单的ID分配器

一个简单的ID分配器，包装了一个整数。可以用于哈希表。

ll06printf committed:
Transformable的基本框架

ll06printf committed:
Transformeable 简单修正

- [fix] 几处笔误鹤分隔符错误
- [add] 一个简单的测试

ll06printf committed:
修正函数名和笔误

- [add] 添加了 `visit_enum` 相关的生成（未测试）
- [modify] 将函数名中的 `function` 变更为 `fn`

ll06printf committed:
更改 Transformable 实现

- [add] Transformable 现在可以用于拥有多个分量的元组变体。

ll06printf committed:
新的 Transformable 的测试

- [add] 测试 Transformable 的对于多个分量的变体的应用
- [add] Transformer Trait 的内部命名方式变更。

ll06printf commented:
提供了对于具有多个分量的元组变体的支持。

l3773l commented on lib/util/src/id_allocator.rs:
```diff
@@ -0,0 +1,66 @@
+        }
+
+        #[test]
+        fn id_allocator_never_repeate() {
```
*repeat

l3773l commented on lib/util/src/id_allocator.rs:
```diff
@@ -0,0 +1,66 @@
+mod sysy_util {
```
好像多套了一层 `sysy_util` 👀

l3773l commented on lib/util/src/lib.rs:
```diff
@@ -0,0 +1 @@
+mod id_allocator;
```
pub

l3773l commented on README.md:
```diff
@@ -1,2 +1,15 @@
+- `ast` AST表示
+- `parser` 解析器
+- `interpreter` 解释器
+- `ast_ir` AST2IR 转换
```
感觉命名上 `ast_to_ir` 或 `ir_emitter` 或 `ast_lowering` 更好一点😵‍💫

l3773l reviewed (approved):
😋

ll06printf committed:
规范性修改

- [modify] `id_allocator` 测试中的局部变量重命名
- [fix] `id_allocator` 模块访问性修正为 pub
- [fix] 移除 `id_allocator` 多余的嵌套模块名。
- [modify] `ast_ir` 子库被重命名为 `ast_to_ir`。
